gfx/drivers/ps2_gfx.c

```diff
@@ -276,7 +276,11 @@ static bool ps2_gfx_frame(void *data, const void *frame,
       bool sendPalette = false;
       struct retro_hw_ps2_insets padding = empty_ps2_insets;
       if (frame != RETRO_HW_FRAME_BUFFER_VALID){ /* Checking if the transfer is done in the core */
-         transfer_texture(ps2->coreTexture, frame, width, height, ps2->PSM, ps2->core_filter, 1);
+         /* calculate proper width based in the pitch */
+         int bytes_per_pixel = (ps2->PSM == GS_PSM_CT32) ? 4 : 2;
+         int real_width = pitch / bytes_per_pixel; 
+         transfer_texture(ps2->coreTexture, frame, real_width, height, ps2->PSM, ps2->core_filter, 1);
```

Comment:
You're not storing the pitch, so this isn't actually "supporting" it... you need to effectively skip `pitch - width` bytes at the end of every line if there are any.

Also I noticed the code in this file does not adhere to RA's style, nor C89, so we need to fix that as well.

Reply:
@fjtrujy explained his reasons for the pitch here -

```
it's working already
mgba use just the half of the screen
the right part is totally dark
in the ps2 I just see full screen mgba
which mean that has been implemented
I'm going to give you 2 links
https://github.com/fjtrujy/RetroArch/blob/master/gfx/drivers/ps2_gfx.c#L280-L283
GitHub
fjtrujy/RetroArch

this is the part that I have added
if you see in the line 283, I'm setting the padding.righ, what is exactly the meaning of the pitch
then later I'm using it here
https://github.com/fjtrujy/RetroArch/blob/master/gfx/drivers/ps2_gfx.c#L138-L173
GitHub
fjtrujy/RetroArch

so this is the function that just print the buffer in the screen, having into consideration the padding
actually that function is amazing, because I can easily manage overscan, without iterate in the buffer itself
it saves a lot of cpu time
```

I think we can merge this as-is, if there are any improvements to be made, they can be done later. The focus right now is on providing the mgba core on the buildbot for users.

I will clean up the code style after this merge, and we will also need to look at making a docs page where we explain code styling guidelines.

Reply:
Alright, I missed that somehow. Apologies @fjtrujy, you had it right.